File: ScpProject/StrengthConApp/src/app/Models/SetsAndReps/WorkoutDetails.ts
import { Week } from "../Week";
import { ITaggable } from '../../Interfaces/ITaggable';
import { Tag } from '../Tag';

export class WorkoutDetails implements ITaggable {
    public Id: number;
    public Name: string;
    public Notes : string;
    public CreatedUserId: number;
    public CreatedDateTime: Date;
    public TotalWorkout: Week[] = [];
    public Rest : string;
    Tags: Tag[] = [];
    public CanModify : boolean = true;
    public ShowRepsAchievedBox: boolean = false;
    public ShowOtherBox: boolean = false;
    public ShowRestBox: boolean = false;
    public ShowDistanceBox: boolean = false;
    public ShowTimeBox: boolean = false;
    public ShowWeight: boolean = true;
    public ShowRepsBox: boolean = true;
    public ShowSetsBox: boolean = true;
    public ShowPercentageBox: boolean = true;
    public ShowAdvancedOptions: boolean = false;

}

File: ScpProject/BL/BusinessObjects/SetsAndReps/WorkoutDetails.cs
using System;
using System.Collections.Generic;

namespace BL.BusinessObjects.SetsAndReps
{
    public class WorkoutDetails
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public String Notes { get; set; }
        public int CreatedUserId { get; set; }
        public string Rest { get; set; }
        public bool ShowWeight { get; set; }
        public virtual List<Week> TotalWorkout { get; set; }
        public List<Tag> Tags { get; set; }
        public Boolean CanModify { get; set; }
        public Boolean IsDeleted { get; set; }
    }
}
